### ISSUE TITLE
Add localestorage support

### ISSUE THREAD

**Issue by Belphemur** [Belphemur/node-json-db#22]
To be able to use it in browser easily and rely on the storage mechanism there.

- [ ]  Create a storage "interface"

- [ ]  Implement driver for hard storage (current one)

- [ ]  Implement storage for LocalStorage

**Comment by ZaneHannanAU** [comment]
Currently not possible without a rework (`requires('')` compatibility required for current implementation).

I'm working on it though, as one with `Promise`s.

Not published, but see https://github.com/ZaneHannanAU/db-dot-json for more information.

**Comment by Belphemur** [comment]
@ZaneHannanAU Great work on implementing the Promises :)

I admit I gave up on Async as it was overkill for the usage I had of the lib at the time (hence the decision to use Sync filesystem calls).

**Comment by ZaneHannanAU** [comment]
@Belphemur The biggest issue right now is that it doesn't work at all.

As in, not even the tests work lol

**Comment by Belphemur** [comment]
@ZaneHannanAU Yep, Promises are a double edge sword. One that I didn't wanted to impose on the user.

Also at the time I started the lib, there were easily 5-6 differents promise lib...

**Comment by Drarig29** [comment]
I think an easier solution would be to:

- Make the file persistance optional (for example by allowing `null` as a database filename in the `JsonDB` constructor)
- Add a public method exposing the private field named `data`
- Let the user make his own solution (either outside of the library, or by passing a save callback function) to persist data manually
- Persisting in files would remain the default behavior
